lib/macho/macho_file.rb

```diff
@@ -409,16 +409,25 @@ def add_rpath(path, _options = {})
     #  file.delete_rpath("/lib")
     #  file.rpaths # => []
     # @param path [String] the runtime path to delete
-    # @param _options [Hash]
+    # @param options [Hash]
+    # @option options [Boolean] :uniq (false) if true, also delete
+    #  duplicates of the requested path. If false, delete the first
+    #  instance (by offset) of the requested path.
     # @return void
     # @raise [RpathUnknownError] if no such runtime path exists
-    # @note `_options` is currently unused and is provided for signature
-    #  compatibility with {MachO::FatFile#delete_rpath}
-    def delete_rpath(path, _options = {})
-      rpath_cmd = command(:LC_RPATH).find { |r| r.path.to_s == path }
-      raise RpathUnknownError, path unless rpath_cmd
+    def delete_rpath(path, options = {})
+      uniq = options.fetch(:uniq, false)
+      search_method = uniq ? :select : :find
+
+      rpath_cmds = command(:LC_RPATH).method(search_method).call { |r| r.path.to_s == path }
+      raise RpathUnknownError, path if rpath_cmds.nil? || rpath_cmds == []
```

Comment:
I feel like there should be a nicer way to do this check, but that's escaping me at the moment.

Reply:
If `MachO::LoadCommands::RpathCommand` had an `empty?` instance method (presumably it should always return `true`?), then we could do
```suggestion
      raise RpathUnknownError, path if rpath_cmds.nil? || rpath_cmds.empty?
```
though I don't know if we want to add an instance method there just for this.

Reply:
Or maybe

```suggestion
      raise RpathUnknownError, path if rpath_cmds.nil? || (rpath_cmds.is_a? Array && rpath_cmds.empty?)
```

I might be overthinking this 😄

Reply:
This might be a little _too_ magical :slightly_smiling_face: 

Maybe something like this?

```ruby
rpath_cmds = if uniq
    [command(:LC_RPATH).find { |r| r.path.to_s == path }]
  else
    command(:LC_RPATH).select { |r| r.path.to_s == path }
  end
```

That's slightly less DRY, but it avoids the type pain.

Reply:
...so I doubled down on the magic, but I think it came out quite cleanly after 😄

Let me know if you'd still rather have two versions of the body of this method depending on whether `uniq` is true.

Reply:
Any thoughts here, @woodruffw? If you still think it's too magical we can split this up more clearly with `if uniq`.

Reply:
Works for me! TIL that you can use `Kernel#Array` to turn scalars into arrays.